app/services/repository_actions/repository_cell_resolver.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module RepositoryActions
+  class RepositoryCellResolver
+    def initialize(cell, new_row, team)
+      @cell    = cell
+      @new_row = new_row
+      @team    = team
+    end
+
+    def call
+      case @cell.value_type
+      when 'RepositoryListValue'
+        clone_repository_list_value
+      when 'RepositoryTextValue'
+        clone_repository_text_value
+      when 'RepositoryAssetValue'
+        clone_repository_asset_value
+      when 'RepositoryDateValue'
+        clone_repository_date_value
+      end
+    end
+
+    private
+
+    def clone_repository_list_value
+      old_value = @cell.value
+      RepositoryListValue.create(
+        old_value.attributes.merge(id: nil,
+          repository_cell_attributes: {
+            repository_row: @new_row,
+            repository_column: @cell.repository_column
+          })
+      )
+    end
+
+    def clone_repository_text_value
+      old_value = @cell.value
+      RepositoryTextValue.create(
+        old_value.attributes.merge(id: nil,
+          repository_cell_attributes: {
+            repository_row: @new_row,
+            repository_column: @cell.repository_column
+          })
+      )
+    end
+
+    def clone_repository_asset_value
+      old_value = @cell.value
+      new_asset = create_new_asset(old_value.asset)
+      RepositoryAssetValue.create(
```

Comment:
I think that `created_by` and `last_modified_by` should be set to the user who is performing the copying of the row.